core/deployment/src/main/java/io/quarkus/deployment/util/WebJarUtil.java

```diff
@@ -344,20 +344,34 @@ private static InputStream insertVariables(ResolvedDependency appArtifact, Input
         if (filename.endsWith(CSS)) {
             Config c = ConfigProvider.getConfig();
             String contents = new String(IoUtil.readBytes(is));
-            contents = contents.replace("{applicationName}",
-                    c.getOptionalValue("quarkus.application.name", String.class)
-                            .orElse(appArtifact.getArtifactId()));
 
-            contents = contents.replace("{applicationVersion}",
-                    c.getOptionalValue("quarkus.application.version", String.class)
-                            .orElse(appArtifact.getVersion()));
+            String applicationName = c.getOptionalValue("quarkus.application.name", String.class)
+                    .orElse(appArtifact.getArtifactId());
+
+            String applicationVersion = c.getOptionalValue("quarkus.application.version", String.class)
+                    .orElse(appArtifact.getVersion());
+
+            contents = replaceHeaderVars(contents, applicationName, applicationVersion);
+
+            contents = contents.replace("{applicationHeader}", getUIHeader(c, applicationName, applicationVersion));
```

Comment:
Sorry for being picky but the pattern you use is a bit weird. I think you should replace `{applicationHeader}` first and only then call `replaceHeaderVars()` and so remove the call to `replaceHeaderVars()` in `getUIHeader()`.

And probably rename `replaceHeaderVars()`.

But I'm not entirely sure you want to replace the header vars in all the contents anyway.